program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+
+Currently LLVM does not support CORE-V specific features or extensions, and subsequently, its test suites do not include any CORE-V or PULP-specific tests.
+
+Other toolchains available for RISC-V are:
+- the GNU tool chain for RISC-V, for which a CORE-V specific project is being set up within OpenHW Group;
+- IAR RISC-V compiler
+
+The GNU tool chain project within OpenHW Group intends to support CORE-V and PULP extensions, but at the time of writing focuses solely on the CV32E40P platform.
+
+An OHG architecture project devoted to the CVA6 (Ariane) family of cores has passed the PPL gate.  There is a clear synergy between this project and the CVA6 project:
+- CVA6 hardware design and verification will directly benefit from the LLVM work;
+- in return, the CVA6 project will directly provide inputs (specifications, documentation) for the compiler- and simulator-related activities in the present project.
+
+## External dependencies
+
+Prequisites:
+- all the upstream tool sources
+- a suitable platform for regression testing the compiler.
+- agreement on the instruction set encodings to be compliant with the RISC-V standard.
+
+External dependencies:
+- ongoing tracking of upstream LLVM development until the CORE-V tool chain is accepted upstream.
+
+## List of project outputs (deliverables)
+
+## Project deliverables
+
+### Final deliverables
+
+1. extensions to upstream LLVM compiler tools to support CORE-V;
+2. revisions to the CORE-V design specifications to clarify ambiguities.
+
+### Interim deliverables
+
+1. Reports on progress to the monthly SW TG:
+   - progress against work packages;
+   - regression test results;
+   - updates to the project plan; and
+   - updates to the risk register.
+2. Continuously updated source code as new features are added.
+
+## TGs impacted/resource requirements
+
+The Software TG will be responsible for oversight of the planning and delivery of this project.
+
+## OpenHW engineering staff resource plan: requirement and availability
+
+- Duncan Bees - program management oversight
+
+## Engineering resource supplied by members - requirement and availability
+
+TBD
+
+*Note:* This is effort by experienced LLVM compiler tool chain / OCaml specialist engineers.
+
+## OpenHW marketing resource - requirement and availability
+
+- press release support if/when CORE-V support is accepted upstream.
+
+## Marketing resource supplied by members - requirement and availability
+
+- No resource requirements yet identified
+
+## Funding supplied by OpenHW - requirement and availability
+
+- None proposed
+
+## Funding supplied by members - requirement and availability
+
+- An alternative to support in kind is funding of effort by Thales and Embecosm.
+
+## Architecture diagram
+
+The standard LLVM tool chain components are shown in the following diagram.
+
+![](images/gnu-tools.png)
```

Comment:
You'll need a LLVM equivalent. Does this help?
![llvm-tools](https://user-images.githubusercontent.com/1662021/99826747-b17bf100-2b50-11eb-9386-76b8139dcdfe.png)